### PR TITLE
Migrate test-coverage report from Travis to GitHub Action

### LINKED ISSUE #17014: Add test coverage report in GitHub action and remove travis in CI
Hi, the travis CI is slower than GitHub action, it is better to remove it and use GitHub action only.
But the test coverage report is generated by travis now, please move it into GitHub action and and remove travis as well.

### PR DESCRIPTION
**Pull request by Jacob953**
Fixes #17014 .

Changes proposed in this pull request:
- add test-coverage report to ci.yml
- delete .travis.yml

Additional 
- codecov report example https://codecov.io/github/Jacob953/shardingsphere/commit/7f83acc714b6f32725a447df6d010b60d5a1f900
- screenshot
image

### CODE REVIEW

**Comment by Jacob953** [comment]
I got a situation in Set Up MySQL here. https://github.com/apache/shardingsphere/actions/runs/2325218817

I'll try Azure/mysql first. If there is a more standard way please let me know. ;)

**Comment by Jacob953** [comment]
I'm stuck. (

There are serval Prerequisites to use Azure/mysql.
- An Azure account with an active subscription.
- A GitHub account. If you don't have a GitHub account.
- A GitHub repository with sample data (data.sql).
- An Azure Database for MySQL server.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -186,3 +186,33 @@ jobs:
...
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+#      - name: Shutdown Ubuntu MySQL(SUDO)
+#        run: sudo service mysql stop
+#      - name: Set up MySQL
+#        uses: mirromutth/mysql-action@v1.1
+#        with:
+#          mysql root password: "root"
```
Comment (terrymanu) [inline .github/workflows/ci.yml:209]: Can we remove the commented lines?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -186,3 +186,33 @@ jobs:
...
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: 8
+      - name: Test with Maven
+        run: ./mvnw -B --no-transfer-progress clean install cobertura:cobertura -Dmaven.javadoc.skip=true -Ddatabases=MySQL,PostgreSQL
```
Comment (terrymanu) [inline .github/workflows/ci.yml:216]: `-Ddatabases=MySQL,PostgreSQL` is for IT only, could we consider about remove the arguments?

**Comment by Jacob953** [comment]
Hi, @terrymanu PTAL! This PR could be merged. ;)

I've tested it on my own repo https://github.com/Jacob953/shardingsphere/pull/5.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**